lib/unsupported-features/node-builtins-modules/test.js

```diff
@@ -38,8 +38,11 @@ const test = {
         setResolveSnapshotPath: { [READ]: { experimental: ["22.3.0"] } },
     },
     MockFunctionContext: { [READ]: { supported: ["19.1.0", "18.13.0"] } },
-    MockModuleContext: { [READ]: { experimental: ["22.3.0"] } },
-    MockTracker: { [READ]: { supported: ["19.1.0", "18.13.0"] } },
+    MockModuleContext: { [READ]: { experimental: ["22.3.0", "20.18.0"] } },
+    MockTracker: {
+        [READ]: { supported: ["19.1.0", "18.13.0"] },
+        module: { [READ]: { experimental: ["20.18.0"] } },
+    },
```

Comment:
I had to test this one because I think I may have misunderstood this one :thinking:

```ts
$ volta run --node 20.18.0 node
Welcome to Node.js v20.18.0.
Type ".help" for more information.
> const test = require('node:test')
undefined
> test.mock.module
undefined
> test.mock.<tab>
test.mock.__proto__             test.mock.hasOwnProperty        test.mock.isPrototypeOf
test.mock.propertyIsEnumerable  test.mock.toLocaleString        test.mock.toString
test.mock.valueOf

test.mock.constructor           test.mock.fn                    test.mock.getter
test.mock.method                test.mock.reset                 test.mock.restoreAll
test.mock.setter                test.mock.timers
```


I think I borked this one originally... As I also dont see any of the classes at all :grimacing:

```js
import * as Test from 'node:test';
console.info(process.version);
console.info(Object.getOwnPropertyNames(Test));
```

```
$ node test.js
v22.11.0
[
  'after',   'afterEach',
  'before',  'beforeEach',
  'default', 'describe',
  'it',      'mock',
  'only',    'run',
  'skip',    'suite',
  'test',    'todo'
]
```

Reply:
According to the docs, these functions are on the `MockTracker` instance you automatically get in your tests

Reply:
Yup, I agree!

---

The issue is that `test.MockTracker` does not get exposed 🤔 

MockTracker is accessable as `test.mock`, and as a property of `TestContext`

`MockTracker` needs to be moved to its own:
```js
const MockTracker = {
  fn: { [READ]: { supported: [ "19.1.0", "18.13.0" ] } },
  getter: { [READ]: { supported: [ "19.3.0", "18.13.0" ] } },
  method: { [READ]: { supported: [ "19.1.0", "18.13.0" ] } },
  module: { [READ]: { experimental: [ "20.18.0" ] } },
  reset: { [READ]: { supported: [ "19.1.0", "18.13.0" ] } },
  restoreAll: { [READ]: { supported: [ "19.1.0", "18.13.0" ] } },
  setter: { [READ]: { supported: [ "19.3.0", "18.13.0" ] } },
}
```

Then referenced like:
```js
const test = {
  // ...
  mock: MockTracker,
};
```

Also, I think these need to be removed:
- `MockFunctionContext`
- `MockModuleContext`
- `MockTracker`
- `MockTimers`
- `TestsStream`
- `TestContext`
- `SuiteContext`

What do you think?

Reply:
Let's add all the other methods + extract the other globals in a separate PR

Reply:
And let's discuss the best approach moving forward on Discord